### PR TITLE
[RC] V0.4

### PR DESCRIPTION
**Pull request by darkdrgn2k**
To fix

- [x] ./install2: line 18: warning: command substitution: ignored null byte in input (install2)
- [x] E: Unable to locate package armbian-firmware-full - pipe to dev null
- [x] IPSF spelled wrong on dialog
- [x] sed: cannot rename /tmp/sedKGXeLi: Operation not permitted [Possible solution](http://helpdoco.com/Linux-Unix/sed-cannot-rename-permission-denied.htm)
- [x] `for wlan in $(find /sys/class/net -maxdepth 1 -name 'wlan*' | xargs -0 basename); do` hostap fix
- [x] /var/www/html/cgj-bin/lib files incorrectly copy into /var/www/html/cgj-bin
- [x] Make selection screen a little bigger
- [x] Yggdrasil detection when not installed or errored out on map
- [x] Yggdrasil detection for status script when not installed

# Tested Install
- [x] Raspberry Pi 3 B - Full Node
- [x] Raspberry Pi 1 B - Basic Node
- [x] Orange Pi Zero 256 - Full Node
- [x] Rock64 - Basic Node
- [x] Nanopi Neo2 - Basic Node (Ubuntu)

# v0.4 Changelog
## Documentation
- Created `MODULES.md` documentation file
- Moved module related content from `README.md`
- Added documentation for various modules - more to come!
- Force password change on first login after install on Raspberry Pi

## Build
- Travis CI runs [shellcheck](https://www.shellcheck.net/) on all .sh and install files
- Created functions to minimize duplicate coding in install making install2 more readable
- Created Profiles for quick node configuration during install.

## Modules
- CJDNS version bump
- Added [Yggdrasil](https://yggdrasil-network.github.io/) routing engine
- Added Yggdrasil IPv6 client IPs to access point ([Yggdrasil subnetting](https://yggdrasil-network.github.io/configuration.html#advertising-a-prefix))
- Added nginx welcome page with information about the node
- CJDNS and Yggdrasil live graphs of peers on welcome page
- Status script now work with Yggdrasil
- Added IP forwarding to iptunnel
- Added IPv6 addressing to iptunnel
- Sorted and cleaned up firewall rules
- Pi stream support for Audio Only (SDR)
- Pi stream patches for stability
- IPFS Version bump 0.4.18
- Added version variable and version bump for Grafana
- Added cjdns compiled DEB for arm64

## Hardware
- Meshpoint and Adhoc dropped to 20Mhz 
- Raspbian onboard WiFi now easier to use in adhoc mode
- Added notes on how to mesh onboard Raspberry Pi WiFi
- Added install support for MK802ii

## Other
- Lots of shellcheck suggested cleanup
- Scuttlebot now called ssb-server
- ipfs-swarm.sh now works with Yggdrasil too
- Added up forwarding rules for Yggdrasil client IPs
- Moved cjdns-setup from systemd rules to post startup script
- More IPFS bootstraps for Hyperborea and Yggdrasil
- IPFS  - enabled gossipsub and filestore 

## Contrib
- Added a port 80 redirect so non-Internet accessible nodes can show their Welcome page
    - Basically a captive gateway
- Added script to build SDR drivers for pi stream

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,43 +39,16 @@ Many board that run [Armbian](https://www.armbian.com/) such as many models of O
...
-| `WITH_EXTRA_TOOLS`              | None                                           | Set to `true` if you want to install non-essential tools useful for network analysis: vim socat oping bmon iperf3 |
-| `WITH_WATCHDOG`                 | None                                           | Set to `true` if you want to enable hardware watchdog that will reset the device when the operating system becomes unresponsive. |
-| `WITH_YRD`                      | None                                           | Set to `true` if you want to enable [yrd](https://github.com/kpcyrd/yrd), a helpful command-line tool for cjdns. |
-
-
-If you are connected to the WiFi Access Point, all HTTP services are available via `http://10.0.0.1:PORT` as well as the cjdns IPv6. To connect with the cjdns address, first note your node's fc00::/8 address from `status`, then navigate to `http://[fcaa:bbbb:cccc:dddd:eeee:0000:1111:2222]:PORT` from your browser.
+During the installation, you may be able to pick a profile, or choose between many modules. To learn what each module is for, look at [MODULES.md](./docs/MODULES.md). This is important for the installation.
```
Comment (darkdrgn2k) [inline README.md]: > you may be able to pick a profile

Makes it sound like there is something stopping you from doing so.

I think this line needs to be re-worded

File: README.md [hunk 2]
```diff
@@ -39,43 +39,16 @@ Many board that run [Armbian](https://www.armbian.com/) such as many models of O
...
 
 ## Check Status
 
 1. Give the Pi about 15 seconds to reboot and SSH back into it. You should find the status of your mesh node automatically printed. You can also print this anytime by running `status`.
 
-1. Verify that **cjdns Service** is active, and **Mesh Interface** (if applicable). The **NODE** section should display a single IPv6 address, that's the identity of your Pi in the cjdns mesh. The **PEERS** section should indicate a list of IPv6 addresses that are active peers to your node. This list will be empty, until you have another nearby node with the same set up.
+2. Verify that **cjdns Service** is active, and **Mesh Interface** (if applicable). The **NODE** section should display a single IPv6 address, that's the identity of your Pi in the cjdns mesh. The **PEERS** section should indicate a list of IPv6 addresses that are active peers to your node. This list will be empty, until you have another nearby node with the same set up.
```
Comment (darkdrgn2k) [inline README.md]: > The **NODE** section should display a single IPv6 address

No longer accurate with Yggdrassil installed.

However cjdns is still the address we name the router after.

File: README.md (hunk 1 above)
Comment (makeworld-the-better-one) [reply README.md]: "You will have the option to pick a profile"?
